frontend/src/metabase/public/lib/types.ts

```diff
@@ -13,9 +13,13 @@ export type EmbedResourceParameter = {
   name: string;
   slug: string;
   type: string;
+  required?: boolean;
+  default?: unknown;
 };
 
-export type EmbeddingParameters = Record<string, string>;
+export type EmbeddingParameterVisibility = "disabled" | "enabled" | "locked";
+
+export type EmbeddingParameters = Record<string, EmbeddingParameterVisibility>;
```

Comment:
This is copied from @WiNloSt's PR https://github.com/metabase/metabase/pull/37617/files#diff-e2b89f462d7fdf5e21ca1aff5a1d91ee203cd086e70c5937246a2c19ec8621e0 and discussed with him.